lib/pure/uri.nim

```diff
@@ -321,6 +321,31 @@ proc `$`*(u: Uri): string =
     result.add("#")
     result.add(u.anchor)
 
+proc reverse*(uri: Uri): string =
+  ## returns the `reverse domain name notation 
+  ## <https://en.wikipedia.org/wiki/Reverse_domain_name_notation>`_. 
+  ## of the given hostname.
+  ## A high performance version of this:
+  ## .. code-block::
+  ##   return hostname.split(".").reversed().join(".")
+  result = ""
+  var
+    pos: int = 0
+    buf: string = ""
+    ch: char
+  while true:
```

Comment:
Use ``for ch in uri.hostname`` instead of this while loop + ``pos`` + ``ch`` variables.